### DISCUSSION TITLE
SpecificPartitionMapping between static-partitioned asset and multi-partitioned asset

### DISCUSSION THREAD

**Discussion by dagsir** [dagster-io/dagster#17757] [Q&A]
Hi all, confused about out how to deal with a partition-mapping scenario. I have an upstream asset: say `asset_multi` that is multi-dimensioned (one dim is dynamic and one is static). I have another asset with another static single dimension (`asset_single`). I want to create a downstream asset with a multi dimension where one dimension is identically mapped to `asset_single` and the other dimension is just the multi-dimensioned collapsed to one the way the multi-part key does it (with `|` delimiter).

`asset_multi` keys: &lt;a, 0&gt;, &lt;a, 1&gt;
`asset_single` keys: &lt;x&gt;, &lt;y&gt;

desired downstream asset keys:
`a|0` , `x`
`a|1` , `x`
`a|0` , `y`
`a|1` , `y`

What is the expected way to solve this given we cant have 3-dim multi-parts?

---

*The question was originally asked in [Dagster Slack](https://dagster.slack.com/archives/C01U954MEER/p1699327915290959?thread_ts=1699327915.290959&cid=C01U954MEER).*

**Comment by sryza** [comment]
@clairelin135 do you know if this exists yet? E.g. something like `MultiToSingleDimensionPartitionMapping` but with the ability to set the mapping for one of the dimensions to be a `SpecificPartitionMapping`?

**Reply by clairelin135** [reply]
Unfortunately not, `MultiToSingleDimensionPartitionMapping` currently assumes that the single-dimension partitions def is a dimension of the multipartitions def

It's fairly straightforward to implement this though -- `BaseMultiPartitionMapping` (which `MultiToSingleDimensionPartitionMapping` inherits from) already has the ability to accept partition mappings between specific dimensions. The main open question is what the API of `MultiToSingleDimensionPartitionMapping` would look like
